embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/schema/TelemetryAttributes.kt

```diff
@@ -0,0 +1,26 @@
+package io.embrace.android.embracesdk.arch.schema
+
+import io.embrace.android.embracesdk.internal.spans.toEmbraceAttributeName
+import io.embrace.android.embracesdk.session.properties.EmbraceSessionProperties
+
+/**
+ * Object that aggregates various attributes and returns a [Map] that represents the values at the current state
+ */
+internal class TelemetryAttributes(
+    private val sessionProperties: EmbraceSessionProperties? = null,
+    private val customAttributes: Map<String, String>? = null
+) {
+    private val map: MutableMap<EmbraceAttributeKey, String> = mutableMapOf()
+
+    /**
+     * Return a snapshot of the current values of the attributes set on this as a [Map]
+     */
+    fun snapshot(): Map<String, String> =
+        mutableMapOf<String, String>().plus(customAttributes ?: emptyMap())
```

Comment:
Why is this `mutableMapOf` necessary?

Reply:
haha, it's not. left over from when I was doing `mapKeysTo` so I was merging into an existing map. I will correct.